src/software/thunderscope/thunderscope_main.py

```diff
@@ -257,18 +254,23 @@
         # else, it will be the diagnostics proto
         current_proto_unix_io = tscope.proto_unix_io_map[ProtoUnixIOTypes.CURRENT]
 
-        # different estops use different ports this detects which one to use based on what is plugged in
-        estop_path = (
-            "/dev/ttyACM0" if os.path.isfile("/dev/ttyACM0") else "/dev/ttyUSB0"
+        estop_mode, estop_path = get_estop_config(
+            args.keyboard_estop, args.disable_communication
         )
 
         with RobotCommunication(
-            current_proto_unix_io,
-            getRobotMulticastChannel(args.channel),
-            args.interface,
-            args.disable_estop,
-            estop_path,
+            current_proto_unix_io=current_proto_unix_io,
+            multicast_channel=getRobotMulticastChannel(0),
```

Comment:
This change shouldn't be happening
```suggestion
            multicast_channel=getRobotMulticastChannel(args.channel),
```